src/DataGrid.tsx

```diff
@@ -940,32 +950,40 @@ function DataGrid<R, SR, K extends Key>(
     return isDraggedOver ? selectedPosition.idx : undefined;
   }
 
-  function getSelectedCellProps(rowIdx: number): SelectedCellProps | EditCellProps<R> | undefined {
-    if (selectedPosition.rowIdx !== rowIdx) return;
+  function getSelectedCellProps(rowIdx: number): SelectedCellProps<R, SR> | undefined {
+    if (selectedPosition.rowIdx !== rowIdx) return undefined;
 
     if (selectedPosition.mode === 'EDIT') {
+      const column = columns[selectedPosition.idx];
+      const row = rows[selectedPosition.rowIdx] as R;
+      const colSpan = getColSpan(column, lastFrozenColumnIndex, { type: 'ROW', row });
+
       return {
-        mode: 'EDIT',
-        idx: selectedPosition.idx,
+        selectedCellIdx: selectedPosition.idx,
+        onFocus: handleFocus,
         onKeyDown: handleKeyDown,
-        editorProps: {
-          editorPortalTarget,
-          row: selectedPosition.row,
-          onRowChange: handleEditorRowChange,
-          onClose: handleOnClose
-        }
+        selectedCellEditor: (
+          <EditCell
```

Comment:
It's pushed to an array so we still need a `key` prop here.
Check the jest output.
```suggestion
          <EditCell
            key={column.key}
```